src/pybind/mgr/cephadm/serve.py

```diff
@@ -566,6 +566,8 @@ def virtual_ip_allowed(host):
 
         try:
             all_slots, slots_to_add, daemons_to_remove = ha.place()
+            daemons_to_remove = [d for d in daemons_to_remove if (d.hostname and self.mgr.inventory._inventory[d.hostname].get(
+                "status", "").lower() != "maintenance")]
```

Comment:
if a host is offline, we should also not remove the daemons. something is strage here. we shouldn't need to have a special case for maintenance here.

Reply:
The way the scheduler works, as far as I understand it, any daemon of the given service type it finds that it can't find on any of the candidate hosts (maintenance/offline hosts are not part of candidates) are marked for removal. Therefore daemons on maintenance/offline hosts will always get marked for removal.

https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/schedule.py#L204-L218